pkg/ccl/streamingccl/streamingest/testdata/cutover_after_pause

```diff
@@ -0,0 +1,73 @@
+# Tests that the user can issue a cutover command while the stream ingestion job is paused, and that
+# the stream ingestion job eagerly completes on resumption.
+
+create-replication-clusters
+----
+
+start-replication-stream
+----
+
+let $start as=source-system
+SELECT clock_timestamp()::timestamp::string
+----
+
+exec-sql as=source-tenant
+CREATE TABLE d.x (id INT PRIMARY KEY, n INT);
+----
+
+exec-sql as=source-tenant
+EXPORT INTO CSV 'userfile:///dx' FROM SELECT 42, 42 UNION ALL SELECT 43, 43;
+----
+
+exec-sql as=source-tenant
+IMPORT INTO d.x CSV DATA ('userfile:///dx/export*-n*.0.csv');
+----
+
+let $afterImport as=source-system
+SELECT clock_timestamp()::timestamp::string
+----
+
+wait-until-high-watermark ts=$afterImport
+----
+
+job as=destination-system pause
```

Comment:
I see you're pausing the job. Can users pause replication (ALTER TENANT .. PAUSE) and then cutover? or maybe in that case we don't allow cutover?

Reply:
Yes, a user can issue a cutover command while the job is paused, but the cutover will not be executed until after the job resumes. 

Users typically pause and resume jobs via PAUSE JOB {jobID}, and RESUME JOB {job ID}. See the [docs.](https://www.cockroachlabs.com/docs/stable/pause-job.html#:~:text=To%20pause%20multiple%20jobs%2C%20nest,by%20maxroach%20will%20be%20paused.) `ALTER TENANT ....` is used for job management that's special to the replication job, like issuing a cutover cmd.

Reply:
well, we don't really want non-advanced users to even know there is a job - we want them to only pause the replication (`ALTER TENANT aa PAUSE REPLICATION`) and not the job. (theoretically we can one day, far in the future, hide the replication jobs or disallow pausing those).

I guess I see now why you added the `job pause` commands to the data driven infra. mmm. do you think we should drop those and instead run a sql command to pause replication (the ALTER TENANT thing) instead of PAUSE JOB?
I think that pausing the job directly is also useful because things should work well in that case too.

to clarify, pausing the replication should be the same as pausing the job.

Reply:
aha! I didn't realize `ALTER TENANT aa PAUSE REPLICATION` was a thing! my bad.

> do you think we should drop those and instead run a sql command to pause replication (the ALTER TENANT thing) instead of PAUSE JOB?

Given that `ALTER TENANT aa PAUSE REPLICATION` is semantically the same as `PAUSE JOB {ingestion job}` and that this test requires that we pause the producer job as well, I'm inclined to keep the test as is, (i.e. keep the `job pause` cmds I added). The commands I added also ensure we wait for the pause and resume cmds to take effect, as they are async.

Reply:
maybe add another test where you pause using the ALTER TENANT PAUSE syntax? because that is what users will do. Also, you're saying that you need to pause also the producer job, but things should work well even if we only pause the ingestion job, right? which is the case if we pause tenant replication. So my point is, can you copy paste this test and pause the tenant repl instead of the job?

Reply:
thanks for the chat! we can add that other test with ALTER TENANT PAUSE in another pr, thanks!